### PR TITLE
VP-3611 Fix discount type

### PR DESCRIPTION
**Pull request by Vectorfield4**
https://virtocommerce.atlassian.net/browse/VP-3611

### Proposed of changes
Describe the technical details of realization provided by you.

### Make sure these boxes are checked:
- [x] Check all the changes in github PR - files count (non of them are redundant, have meaningful changes, all are added), if target branch is correct
- [x] Check methods and variable namings - it should be self descriptive, no typos
- [x] Check you did not introduce breaking changes in API and public models/services.
- [x] Respect extensibility - https://community.virtocommerce.com/t/extensibility-basics-the-domain-model-and-persistence-layer-extension/141
- [x] Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [SOLID](https://en.wikipedia.org/wiki/SOLID) principles
- [x] For unit tests - follow Microsoft best practices: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices
- [x] Consolidate solution dependencies in case you are using newer version, update VC module dependencies in module.manifest. Do not upgrade 3rd party packages that are shipped with the platform with the version newer than in the platform.
- [x] Check code style conventions - https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md
- [x] Check PR have a concise and descriptive title, follow [git commit message rules](https://github.com/VirtoCommerce/styleguide/blob/master/gitcommits.md)

### CODE REVIEW

**Comment by Vectorfield4** [comment]
Merge this with https://github.com/VirtoCommerce/vc-module-experience-api/pull/46

File: VirtoCommerce.Storefront.IntegrationTests/Responses/GetProductsByIds.json [hunk 1]
```diff
@@ -872,7 +872,7 @@
       "allowBackorder": false,
       "status": 0,
       "fulfillmentCenterId": "vendor-fulfillment",
-      "inStockQuantity": 7,
+      "inStockQuantity": 6,
```
Comment (yecli) [inline VirtoCommerce.Storefront.IntegrationTests/Responses/GetProductsByIds.json:875]: It is 7 in the sample data, need to revert this one

File: VirtoCommerce.Storefront.IntegrationTests/Responses/GetProductsByIds.json [hunk 2]
```diff
@@ -882,12 +882,12 @@
         "allowBackorder": false,
         "status": 0,
         "fulfillmentCenterId": "vendor-fulfillment",
-        "inStockQuantity": 7,
+        "inStockQuantity": 6,
```
Comment (yecli) [inline VirtoCommerce.Storefront.IntegrationTests/Responses/GetProductsByIds.json:885]: It is 7 in the sample data, need to revert this one

File: VirtoCommerce.Storefront.IntegrationTests/Responses/GetProductsByIds.json [hunk 3]
```diff
@@ -882,12 +882,12 @@
...
         "reservedQuantity": 0,
         "productId": "9cbd8f316e254a679ba34a900fccb076"
       }
     ],
-    "availableQuantity": 7,
+    "availableQuantity": 6,
```
Comment (yecli) [inline VirtoCommerce.Storefront.IntegrationTests/Responses/GetProductsByIds.json:890]: It is 7 in the sample data, need to revert this one

**Pull request merged**